linksight/api/models.py

```diff
@@ -82,15 +83,15 @@ def generate_match_items(self, **kwargs):
         with self.dataset.file.open() as f:
             dataset_df = pd.read_csv(f)
 
-        matches = get_matches(dataset_df, columns=self.loc_columns)
+        matches = NgramsMatcher().get_matches(dataset_df, columns=self.loc_columns)
```

Comment:
`dataset_df` and `columns` might be common arguments for the matcher. Maybe move them to `BaseMatcher.__init__`?

Reply:
Originally, I thought of passing the dataset's file path to the matcher instead of it being a pandas dataframe. Since our output is no longer a dataframe, I think we can go with that route so whoever is doing the matcher does not necessarily have to use Pandas

Reply:
Or an iterable! So the matcher doesn't do IO aside from (maybe) loading its reference file.